src/parse/lex.rs

```diff
@@ -1,22 +1,319 @@
+}
+
+///checks for whitespace/line break/tab
+fn is_whitespace(c: char) -> bool {
+    match c {
+        ' ' | '\n' | '\t' => true,
+        _ => false
+    }
+}
+
+impl<'a> Iterator for Lexer<'a> {
+
+    type Item = TokenSpan;
+
+    fn next(&mut self) -> Option<TokenSpan> {
+
+        //for two char ops (e.g. >=), we need to check the next char
+        //if we are at the end of the string, we end
+        let nexchar = self.next.unwrap_or('\x00');
+
+        //saving the first pos of a token
+        self.span_start = self.curr_pos;
+
+        //if no current char available, return None
+        if self.curr.is_none() {
+            return None;
+        }
+
+        //unwrap current char and decide token class
+        let token = match self.curr.unwrap_or('x') {
+
+            //words
+            'a' ... 'z' | 'A' ... 'Z' => {
+                let w = self.scan_words();
+                Token::Word(w.to_lowercase())
+            },
+
+            //nums
+            '0' ... '9' => {
+                let n = self.scan_nums();
+                Token::Num(n)
+
+            },
+
+            //semi
+            ';' => {
+                self.bump();
+                Token::Semi
+            },
+
+            //dots
+            '.' => {
+                self.bump();
+                Token::Dot
+            },
+
+            //commas
+            ',' => {
+                self.bump();
+                Token::Comma
+            },
+
+            // //bang
+            // '!' => {
+            //     self.bump();
+            //     Token::Bang
+            // },
+
+            // //question marks
+            // '?' => {
+            //     self.bump();
+            //     Token::QMark
+            // },
+
+            //ParenOp
+            '(' => {
+                self.bump();
+                Token::ParenOp
+            },
+
+            //ParenCl
+            ')' => {
+                self.bump();
+                Token::ParenCl
+            },
+
+            //Literals
+            '\'' | '"' => {
+                let l = self.scan_lit();
+                self.bump();
+                Token::Literal(Lit::Str(l))
+            },
+
+            //Equ
+            '=' => {
+                self.bump();
+                Token::Equ
+            },
+
+            //GEThan >=
+            '>' if nexchar == '=' => {
+                self.dbump(); //because two chars!
+                Token::GEThan
+            },
+
+            //SEThan <=
+            '<' if nexchar == '=' => {
+                self.dbump();
+                Token::SEThan
+            },
+
+            //GThan >
+            '>' => {
+                self.bump();
+                Token::GThan
+            },
+
+            //SThan <
+            '<' => {
+                self.bump();
+                Token::SThan
+            },
+
+            //NEqu !=
+            '!' if nexchar == '=' => {
+                self.dbump();
+                Token::NEqu
+            },
```

Comment:
Apparently the `!=` operator was added by some database servers without it being ANSI compliant (see [here](http://stackoverflow.com/a/723317/2408867)). The original "not equal" operator in SQL is `<>`...